### DISCUSSION TITLE
HSS to select flash mode by checking GPIO state instead of usbdmsc hss cli app

### DISCUSSION THREAD

**Discussion by govindsi** [polarfire-soc/polarfire-soc-documentation#367] [HSS and U-Boot]
I was exploring to to add logic in HSS to  read one GPIO state( controlled via external switch) for selecting flash mode, instead of invoking usbdmsc. I am wondering , if there is already some documentation available to support such requirement(modification in HSS_Init ->RunInitFunctions flow)? One simple implementation could be check flash switch state(via GPIO read ) and start usbdmsc_service from RunInitFunctions flow.

**Comment by vfalanis** [comment]
Hi @govindsi ,

Adding support for GPIO events in the HSS is something we have identified before and we are currently working on. We will update this GitHub discussion post once this feature is available. I would also suggest to subscribe to the PolarFire SoC GitHub organisation to get notifications on updates and releases.

Thanks
